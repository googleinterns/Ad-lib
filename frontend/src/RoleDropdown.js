import React from 'react';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import ListItemText from '@material-ui/core/ListItemText';
import Select from '@material-ui/core/Select';

const roles = [
  'Software Engineer',
  'Product Manager',
  'UI/UX Designer',
  'Recruiter',
  'Intern',
  'VP',
  'Director',
  'CEO'
];

export default function RoleDropdown(props) {
  return (
    <div>
      <FormControl style={{width: 180}}>
        <InputLabel id="role">Role</InputLabel>
        <Select
          id="role-input"
          name="role"
          onChange={event => props.onChange(event.target.value)}
          inputProps={{ "aria-label": "role" }}
        >
          {roles.map((currentRole) => (
            <MenuItem key={currentRole} value={currentRole}>
              <ListItemText primary={currentRole} />
            </MenuItem>
          ))}  
        </Select>
      </FormControl>
    </div>
  );
}