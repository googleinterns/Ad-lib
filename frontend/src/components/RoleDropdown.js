import React from 'react';
import PropTypes from 'prop-types';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import ListItemText from '@material-ui/core/ListItemText';
import Select from '@material-ui/core/Select';

// TO-DO: Add more role options to this array
const roles = [
  'Software Engineer',
  'Product Manager',
  'UI/UX Designer',
  'Recruiter',
  'Intern',
  'VP',
  'Director',
  'CEO',
];

// Add onChange to props validation
RoleDropdown.propTypes = {
  onChange: PropTypes.func,
};

/**
 * Create role dropdown component using roles array
 * @param {Object} props
 * @return {RoleDropdown} RoleDropdown component
 */
export default function RoleDropdown(props) {
  return (
    <div>
      <FormControl style={{width: 180}}>
        <InputLabel id="role">Role</InputLabel>
        <Select
          id="role-input"
          name="role"
          onChange={(event) => props.onChange(event.target.value)}
          inputProps={{'aria-label': 'role'}}
        >
          {roles.map((currentRole) => (
            <MenuItem key={currentRole} value={currentRole}>
              <ListItemText primary={currentRole} />
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </div>
  );
}
