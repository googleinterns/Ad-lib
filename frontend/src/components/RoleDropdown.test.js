import React from 'react';
import RoleDropdown from './RoleDropdown';
import renderer from 'react-test-renderer';

describe('Role Dropdown', () => {
  it('should be defined', () => {
    expect(RoleDropdown).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = renderer.create(<RoleDropdown />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
