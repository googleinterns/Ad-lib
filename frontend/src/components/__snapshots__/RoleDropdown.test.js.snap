// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Role Dropdown should render correctly 1`] = `
<div>
  <div
    className="MuiFormControl-root"
    style={
      Object {
        "width": 180,
      }
    }
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
      id="role"
    >
      Role
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-label="role"
        aria-labelledby="role-input"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="role-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        name="role"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value=""
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
<div>
  <div
    class="MuiFormControl-root"
    style="width:180px;"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      id="role"
    >
      Role
    </label>
    <div class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl">
      <div
        aria-haspopup="listbox"
        aria-label="role"
        aria-labelledby="role-input"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="role-input"
        role="button"
      >
        <span></span>
      </div>
      <input
        aria-hidden
        class="MuiSelect-nativeInput"
        name="role"
        value=""
      >
      <svg
        aria-hidden
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path d="M7 10l5 5 5-5z"></path>
      </svg>
    </div>
  </div>
</div>
`;
